### PR TITLE
feat(core): Bridge node short-circuits storing a historical EDS if pruning is enabled

### PR DESCRIPTION
**Pull request by renaynay**
This PR implements short-circuiting storing EDSes in the case that they are outside the sampling window in the case of a syncing bridge node if `--experimental-pruning` is enabled.

Overrides #3261

### CODE REVIEW

File: core/exchange.go [hunk 1]
```diff
@@ -150,11 +155,14 @@ func (ce *Exchange) Get(ctx context.Context, hash libhead.Hash) (*header.Extende
 			&block.Height, hash, eh.Hash())
 	}
 
-	ctx = ipld.CtxWithProofsAdder(ctx, adder)
-	err = storeEDS(ctx, eh.DAH.Hash(), eds, ce.store)
+	ctx = ipld.CtxWithProofsAdder(ctx, adder) // TODO @renaynay: should we short-circuit this if pruning enabled
```
Comment (renaynay) [inline core/exchange.go:158]: cc @walldiss 

we can make this conditional on whether block is outside avail window

Comment (walldiss) [reply core/exchange.go:158]: Yes

File: core/listener_test.go [hunk 2]
```diff
@@ -79,17 +80,11 @@ func TestListenerWithWrongChainRPC(t *testing.T) {
...
 	fetcher, _ := createCoreFetcher(t, cfg)
 	eds := createEdsPubSub(ctx, t)
 
 	store := createStore(t)
-	err := store.Start(ctx)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		err = store.Stop(ctx)
-		require.NoError(t, err)
-	})
```
Comment (Wondertan) [inline core/listener_test.go:92]: Did this get removed as unnecessary?

**Review by Wondertan — APPROVED** [review]
LGTM and the proofs have to be excluded when outside of the window, as you state

File: core/listener_test.go (hunk 2 above)
Comment (renaynay) [reply core/listener_test.go:92]: Yes, `createStore` manages lifecycle

File: core/exchange.go [hunk 3]
```diff
@@ -190,10 +195,33 @@ func (ce *Exchange) getExtendedHeaderByHeight(ctx context.Context, height *int64
...
+	eh *header.ExtendedHeader,
+	eds *rsmt2d.ExtendedDataSquare,
+	adder *ipld.ProofsAdder,
+) error {
+	if !pruner.IsWithinAvailabilityWindow(eh.Time(), ce.availabilityWindow) {
+		log.Debugw("skipping storage of historic block", "height", eh.Height())
+		return nil
+	}
+
+	ctx = ipld.CtxWithProofsAdder(ctx, adder) // TODO @renaynay: should we short-circuit this if pruning enabled
```
Comment (Wondertan) [inline core/exchange.go:217]: Comment seems irrelevant anymore

**Review by walldiss — APPROVED** [review]
LGTM.

**Review by vgonkivs — APPROVED** [review]

**Review by abstraking — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]
